src/VulkanInfoProvider.cpp

```diff
@@ -1,11 +1,293 @@
+            {"heapIndex", _physicalDeviceInfo->memoryProperties.memoryTypes[i].heapIndex},
+        }
+        );
+    }
+
+    for (size_t i = 0; i < _physicalDeviceInfo->memoryProperties.memoryHeapCount; ++i) {
+        json["memory"]["memoryHeaps"].push_back(
+        {
+            {"size", _physicalDeviceInfo->memoryProperties.memoryHeaps[i].size},
+            {"flags", lug::Graphics::Vulkan::API::RTTI::VkMemoryHeapFlagsToStrVec(_physicalDeviceInfo->memoryProperties.memoryHeaps[i].flags)},
+        }
+        );
+    }
+
+    for (auto tmp : _physicalDeviceInfo->queueFamilies) {
+        json["queues"].push_back({
+            {"queueFlags", lug::Graphics::Vulkan::API::RTTI::VkQueueFlagsToStrVec(tmp.queueFlags)},
+            {"queueCount", tmp.queueCount},
+            {"timestampValidBits", tmp.timestampValidBits},
+            {"minImageTransferGranularity", {
+                {"width", tmp.minImageTransferGranularity.width},
+                {"height", tmp.minImageTransferGranularity.height},
+                {"depth", tmp.minImageTransferGranularity.depth}
+            }}
+        });
+    }
+
+    for (auto extension : _physicalDeviceInfo->extensions) {
+        json["extension"].push_back(
+        {
+            {"extensionName", extension.extensionName},
+            {"specVersion", {
+                {"major", static_cast<uint32_t>(lug::Core::Version::fromInt(extension.specVersion).major)},
+                {"minor", static_cast<uint32_t>(lug::Core::Version::fromInt(extension.specVersion).minor)},
+                {"patch", static_cast<uint32_t>(lug::Core::Version::fromInt(extension.specVersion).patch)}
+            }},
+        }
+        );
+    }
+
+    for (auto format : _physicalDeviceInfo->formatProperties) {
+        json["formats"].push_back(
+        {
+            lug::Graphics::Vulkan::API::RTTI::toStr(format.first),
+            {
+                {"linearTilingFeatures", lug::Graphics::Vulkan::API::RTTI::VkFormatFeatureFlagsToStrVec(format.second.linearTilingFeatures)},
+                {"optimalTilingFeatures", lug::Graphics::Vulkan::API::RTTI::VkFormatFeatureFlagsToStrVec(format.second.optimalTilingFeatures)},
+                {"bufferFeatures", lug::Graphics::Vulkan::API::RTTI::VkFormatFeatureFlagsToStrVec(format.second.bufferFeatures)},
+            }
+        }
+        );
+    }
+
+    //    for (auto surface : _physicalDeviceInfo->swapchain)
+    //  {
+    json["zSwapChainInfo"].push_back(
+    {
+        {"minImageCount", _physicalDeviceInfo->swapchain.capabilities.minImageCount},
+        {"maxImageCount", _physicalDeviceInfo->swapchain.capabilities.maxImageCount},
+        {"currentExtent",
+        {
+            {"width", _physicalDeviceInfo->swapchain.capabilities.currentExtent.width},
+            {"height", _physicalDeviceInfo->swapchain.capabilities.currentExtent.height}
+        }
+        },
+        {"minImageExtent",
+        {
```

Comment:
The indentation is wrong, you need to put the bracket at the end of the previous line.